Tools/LogAnalyzer/DataflashLog.py

```diff
@@ -521,7 +521,7 @@ def process(self, lineNumber, e):
                     self.modeChanges[lineNumber] = (modes[int(e.Mode)], e.ThrCrs)
                 except:
                     self.modeChanges[lineNumber] = (e.Mode, e.ThrCrs)
-            elif self.vehicleType == "ArduPlane" or self.vehicleType == "ArduRover":
+            elif self.vehicleType in ["ArduPlane", "APM:Plane", "ArduRover", "APM:Rover", "APM:Copter"]:
```

Comment:
No, that wouldn't work. I moved it here intentionally to fix AttributeError execptions, as newer Copter releases write a mode line similar to Plane/Rover (no e.ThrCrs attribute, but e.Mode and e.ModeNum).